src/driver/mongodb/MongoQueryRunner.ts

```diff
@@ -6,38 +7,41 @@ import {TableForeignKey} from "../../schema-builder/table/TableForeignKey";
 import {TableIndex} from "../../schema-builder/table/TableIndex";
 import {View} from "../../schema-builder/view/View";
 import {
+    AggregateOptions,
     AggregationCursor,
-    BulkWriteOpResultObject,
+    AnyBulkWriteOperation,
+    BulkWriteOptions,
+    BulkWriteResult,
     ChangeStream,
     ChangeStreamOptions,
-    Code,
     Collection,
-    CollectionAggregationOptions,
-    CollectionBulkWriteOptions,
-    CollectionInsertManyOptions,
-    CollectionInsertOneOptions,
     CollectionOptions,
-    CollStats,
-    CommandCursor,
-    Cursor,
-    DeleteWriteOpResultObject,
-    FindAndModifyWriteOpResultObject,
-    FindOneAndReplaceOption,
-    GeoHaystackSearchOptions,
-    GeoNearOptions,
-    InsertOneWriteOpResult,
-    InsertWriteOpResult,
+    CollStats, CollStatsOptions,
+    CountDocumentsOptions,
+    CreateIndexesOptions,
+    DeleteOptions,
+    DeleteResult,
+    DistinctOptions,
+    DropIndexesOptions,
+    Filter,
+    FindCursor,
+    FindOneAndDeleteOptions,
+    FindOneAndUpdateOptions,
+    IndexDescription,
+    IndexInformationOptions,
+    IndexSpecification,
+    InsertManyResult, InsertOneOptions, InsertOneResult,
+    ListIndexesCursor,
+    ListIndexesOptions, MapFunction,
     MapReduceOptions,
+    ModifyResult,
     MongoClient,
-    MongoCountPreferences,
-    MongodbIndexOptions,
-    OrderedBulkOperation,
-    ParallelCollectionScanOptions,
-    ReadPreference,
-    ReplaceOneOptions,
+    OptionalId,
+    OrderedBulkOperation, ReduceFunction,
+    RenameOptions, ReplaceOptions,
     UnorderedBulkOperation,
-    UpdateWriteOpResult
-} from "./typings";
+    UpdateFilter, UpdateOptions, UpdateResult,
+} from "mongodb";
```

Comment:
If we do this wouldn't we have to define `mongodb` as a full dependency rather than a peer and dev dependency?

Reply:
No not really. At runtime the proper dependency is available, as it comes from the calling application. For compile time we have the one as dev dependency.
The only point is that we have to ensure that the peer dependency matches. But this also happened up to now (which is actually why I started investigating in typeorm).

Reply:
I have a good feeling this will fail during runtime even if we aren't using mongo for cases like the browser at the very least.

>  The only point is that we have to ensure that the peer dependency matches. But this also happened up to now (which is actually why I started investigating in typeorm).

There's a very specific way we do imports of these peer dependencies - using `PlatformTools` - so that we don't bring in the dependencies in cases where it would cause failures.

Reply:
Ah now I now what you mean - it's because the import is evaluated when reading the files. But that would mean that we'd have to copy all the dependency types to typeorm again - that's also weird.

Reply:
> that's also weird.

I 100% agree.